front/src/applications/operationalStudies/components/Scenario/ScenarioContent.tsx

```diff
@@ -216,7 +202,15 @@ const ScenarioContent = ({
               )}
               {isMacro ? (
                 <div className={cx(collapsedTimetable ? 'macro-container' : 'h-100')}>
-                  <NGE dto={ngeDto} onOperation={handleNGEOperation} />
+                  {ngeProps.type === 'loading' && <LoaderFill />}
+                  <NGE
+                    dto={ngeProps.type === 'ready' && ngeProps.data ? ngeProps.data.dto : undefined}
```

Comment:
When `ngeProps.type === 'ready'`, why do we need to check if `ngeProps.data` is null/undefined? It should always be an object, no?

Reply:
At line 82 we have `if (!isMacro) return null;` in the `useAsyncMemo` that's why the result can be `null` even if we have  `type === 'ready'.
I know that at line 203 we have `{isMacro ? ( ....` so this case will never happened, but the compilator don't know it.
I can change the code by `dto={ngeProps.type === 'ready' ? ngeProps.data!.dto : undefined}` if you prefer, but the day we want to change the def of the `useAsyncMemo` the compilator will not help us.

Reply:
Right, `useAsyncMemo()`'s API is a bit cumbersome here. Thanks for the explanation, the current approach sounds fine to me.